orc.js

```diff
@@ -1,42 +1,87 @@
-var app = require('commander');
+var commander = require('commander');
 var git = require('gitty');
 var path = require('path');
-
-// process.stdin.resume();
-// process.stdin.setEncoding('utf8');
-// process.stdin.on('data', function(data) {
-//  process.stdout.write(data);
-// });
+var pushover   = require('pushover');
 
 process.on('SIGINT', function () {
   console.log('Got a SIGINT. Goodbye cruel world');
   process.exit(0);
 });
 
-function _requireCommit() {
-
+function _setup(dir) {
+  // TODO: march up to root dir just to make output consistent
+  var repo = git(dir);
+  branches = repo.getBranchesSync();
+  if (!branches.current) {
+    throw new Error('error: '+repo.name+' doesn\'t appear to be a valid git repo');
+  }
+  return { repo_dir: dir, repo: git(dir)};
 }
 
-function checkpoint() {
+function _requireClean(orc) {
+  var status = orc.repo.statusSync();
+  if(status.unstaged.length > 0) {
+    throw new Error('unstaged files, commit before running this.');
```

Comment:
FB convention is that _ indicates private for member functions. My intention was to convey an internal helper function rather than a top level one. how do you feel about this?